packages/providers/src/utils/deepmerge.utils.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { deepmerge, deepmergeAll } from './deepmerge.utils';
+
+describe('deepmerge', () => {
+  it('should merge two objects', () => {
+    const obj1 = {
+      a: {
+        b: 1,
+        d: {
+          a: 1,
+        },
+        x: [1, 2, 3],
+      },
+    };
+
+    const obj2 = {
+      a: {
+        c: 2,
+        d: {
+          a: 1,
+        },
+        x: [3, 4, 5],
+      },
+    };
+
+    const result = deepmerge(obj1, obj2);
+
+    expect(result).toEqual({
+      a: {
+        b: 1,
+        c: 2,
+        d: {
+          a: 1,
+        },
+        x: [1, 2, 3, 3, 4, 5],
+      },
+    });
+  });
+
+  it('should merge a list of objects', () => {
+    const obj1 = {
+      a: {
+        b: 1,
+        d: {
+          a: 1,
+        },
+        x: [1, 2, 3],
+      },
+    };
+
+    const obj2 = {
+      a: {
+        c: 2,
+        d: {
+          a: 1,
+        },
+        x: [3, 4, 5],
+      },
+    };
+
+    const result = deepmergeAll([obj1, obj2]);
+
+    expect(result).toEqual({
+      a: {
+        b: 1,
+        c: 2,
+        d: {
+          a: 1,
+        },
+        x: [1, 2, 3, 3, 4, 5],
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** ‏Great test! Very easy to read.